pcsx2-qt/Debugger/CpuWidget.cpp

```diff
@@ -1023,17 +1055,46 @@ void CpuWidget::onSearchButtonClicked()
 
 		m_ui.listSearchResults->clear();
 		const auto& results = workerWatcher->future().result();
-
-		for (const auto& address : results)
+		QStringList addressList;
+		for (u32 addressNum : results)
 		{
-			QListWidgetItem* item = new QListWidgetItem(QtUtils::FilledQStringFromValue(address, 16));
-			item->setData(256, address);
```

Comment:
The reason I've done it this way is that `256` is the "User Role" (I probably should've used that enum instead). The User Role (and higher) is where you can store any data in there. In this case I stored the raw numerical address there.
With these changes, double clicking on the memory search result ends up taking you to address 0. It's up to you but you can revert the changes here, or modify the double click listener on line 105 to convert the display role (the text) to a number and use that instead.
In the future this will probably be turned into a QListView instead of a QListWidget, but I digress.